omegaconf/omegaconf.py

```diff
@@ -591,6 +591,9 @@ def to_container(
         :param structured_config_mode: Specify how Structured Configs (DictConfigs backed by a dataclass) are handled.
             By default (`structured_config_mode=SCMode.DICT`) structured configs are converted to plain dicts.
             If `structured_config_mode=SCMode.DICT_CONFIG`, structured config nodes will remain as DictConfig.
+            If `structured_config_mode=SCMode.INSTANTIATE`, this function will instantiate structured configs
+               (DictConfigs backed by a dataclass), by creating an instance of the underlying dataclass.
+               See also OmegaConf.to_object.
```

Comment:
This should require resolve=True.
I don't think we should support instantiate without resolve as it may be used to create malformed objects.
(interpolation type is string and the underlying type may be something else, it could even an interpolation to another object).

Reply:
I see your point, but the dataclasses we use right now to define structured configs can already be "malformed", and they are still useful. So I can imagine someone wanting to use resolve=False (probably to do some manipulation on the dataclass then convert it back into a structured config).

Reply:
Why manipulate the dataclass instead of the config in that case?

Reply:
I don't know, maybe the function doing the manipulation is actually a method of the dataclass? Or the goal is to modify / extend the dataclass in a way that is not supported by the public DictConfig API?
(anyway, I don't actually have a concrete use case I need this for myself, so it's not a huge deal, I just wanted to point out that "malformed" dataclass objects are not necessarily always bad)

Reply:
I think the primary reason to convert to a dataclass instance is to lose all traces of OmegaConf and have a good old dataclass you can use normally.
Having interpolation values left-over seems to go against it.

I am okay with allowing this combination in the (now) lower-level `to_container()`, but I think to_object() should be more opinionated.

Reply:
Makes sense :)